## Using logs to solve an equation with $e^x$ in it

The equation
$V_c(t) = V ( 1 - e^\frac{t}{C R})$

First we substitute in the values from the worksheet 7 question 1.d:

$12 = 24 (1 - e^{\frac{t}{100 ms}})$ 

To solve an equation, you can do whatever you want to it so long as you to it to both sides. The game of algebra is figuring out what to do the solve the equation. In this case, we want an equation in the form of $t = ...$ as we want to know what $t$ is.

We start by dividing both sides by 24 to move it to the left hand side of the equals sign:

$\frac{12}{24} = \frac{24 (1 - e^{\frac{t}{100 ms}})}{24}$ 

as $\frac{24 x}{x} = x$, we get:

$\frac{12}{24} = 1 - e^{\frac{t}{100 ms}}$

which just looks like we've moved the 24 over to the left.

Next, we want to get $t$ down from being an exponent, $e^t$, to $t$. To do that we need to know some rules of logs. 

So, logarithms were invented to multiply and divide numbers without a calculated. These are the rules people have to learn:

|Rule or special case | Formula              |
|---------------------|----------------------|
|Product	          | ln(xy)=ln(x)+ln(y)   |
|**Quotient**	          | **ln(x/y)=ln(x)−ln(y)**  |
|Log of power	      | ln(xy)=yln(x)        |
|**Log of e**             | **ln(e)=1**              |
|**Log of one**	          | **ln(1)=0**              |
|Log reciprocal	      | ln(1/x)=−ln(x)       |

We will use quotient, log of e and log of 1.

Next, we take logs, and as we are working in base $e$ we must use the natural logarithm, $ln$ (it stands for Logarithm Natural). Thus:

$\frac{12}{24} = 1 - e^{\frac{t}{100 ms}}$ 

take logs boths sides to get:

$ln(\frac{12}{24}) = ln(1 - e^{\frac{t}{100 ms}})$

using quotient rule on the left handside we have $ln(12/24) = ln(12) - ln(24)$
giving:

$ln(12) - ln(24) = ln(1 - e^{\frac{t}{100 ms}})$

On the right hand side we can expand that bracket, thus:

$ln(12) - ln(24) = ln(1) - ln(e^{\frac{t}{100 ms}})$

We know that $ln(1)$ is 0, so we can substitute that in. First, below, I'll check that 4ln(1) = 0$

I'm writing this in a programming document so it will look different to what you get. You need to use a calculator.

In [1]:
from numpy import log as ln
ln(1)

0.0

So:
$ln(12) - ln(24) = 0 - ln(e^{\frac{t}{100 ms}})$


Now, we know that the $ln$ is the inverse of $e$, so 

$ln(e^x) = x$

and this is what we use to move the $t$ down from being an exponent $e^t$ so, the right hand side of our equation is:

$ln(12) - ln(24) = 0 - \frac{t}{100 ms}$
Now we should probably calculate the logs on the left hand side:

You can do either $ln(12)-ln(24)$ or $ln(12/25)$ or $ln(0.5)$

In [2]:
ln(12)

2.4849066497880004

In [3]:
ln(24)

3.1780538303479458

In [4]:
ln(12)-ln(24)

-0.6931471805599454

In [5]:
ln(12/24)

-0.6931471805599453

In [6]:
ln(0.5)

-0.6931471805599453

So, we now have 

$-0.6931471805599453 = 0 - \frac{t}{100 ms}$

Now, we want $t$ on it's own, and it is divided by 100, so we multiply both sides by 100. Note that we can do this as $0 = 0/100$ as 0 divided by any number is still 0. 

So, we could write our equation as

$-0.6931471805599453 = \frac{0}{100} - \frac{t}{100 ms}$ 

then we multiply both sides by 100 and as $\frac{100}{100} = 1$ because any number divided by itself is 1.

$100 \times -0.6931471805599453 = 100\times\frac{0}{100}  - 100 \times \frac{t}{100 ms}$

gives


In [7]:
-0.6931471805599453 * 100

-69.31471805599453

$-69.31471805599453 = 0 -  t$

$0 - t$ is just $-t$ so

$-69.31471805599453 = -  t$

we wanted an equation 't = ...' something and we are nearly there. We can switch the left and right hand sides (they're still equal, so we can do this)

$-t = -69.31471805599453 $

Now to get rid of the minus sign we need to multiply by -1 as $-x \times -1 = x$, and $x \times -1 = -x$

so

$-t \times -1 = -69.31471805599453 \times -1$

finally:

$t = 69.31471805599453$

and then check the units, and round:

$t = 69.314 ms$

Done!



Note, you wouldn't usually write all these steps out in detial, you usually just write 'multiply b.s. by 100' and then do it

$12 = 24 (1 - e^{\frac{t}{100 ms}})$   divide b.s. 24

$\frac{12}{24} =  1 - e^{\frac{t}{100}}$  take logs

$ln(\frac{12}{24}) =  ln(1) - ln(e^{\frac{t}{100 ms}})$

$ln(0.5) =  0 - \frac{t}{100}$   multiply b.s. by 100

$100 \times ln(0.5) = - t$  multiply b.s. by -1

$t = 100 \times ln(0.5)$

$t = 69.314ms$


Also, you may see $Log_{10}$ which is log in base 10, used if you have numbers like $10^x$. Or Log_2 which is log in base 2, used for binary. 